src/agent/actions/transform.ts

```diff
@@ -0,0 +1,135 @@
+import {type Observable} from 'rxjs'
+
+import {_request} from '../../data/dataMethods'
+import type {ObservableSanityClient, SanityClient} from '../../SanityClient'
+import type {AgentActionParams, Any, HttpRequest, IdentifiedSanityDocumentStub} from '../../types'
+import {hasDataset} from '../../validators'
+import type {
+  AgentActionAsync,
+  AgentActionPathSegment,
+  AgentActionRequestBase,
+  AgentActionSync,
+  AgentActionTarget,
+  AgentActionTargetInclude,
+} from './commonTypes'
+
+/** @beta */
+export interface TransformRequestBase extends AgentActionRequestBase {
+  /** schemaId as reported by sanity deploy / sanity schema store */
+  schemaId: string
+
+  /**
+   * The source document the transformation will use as input.
+   */
+  documentId: string
+
+  /**
+   * The target document will first get content copied over from the source,
```

Comment:
```suggestion
   * The source document's content is first copied to the target,
```
It's a little confusing here to say 'the target document' as it kind of implies that the target is being copied, rather than the source (`documentId`) above.